### PR TITLE
chore(sidebar): reorganize everything under `use kuma`

### PR DESCRIPTION
**Pull request by lahabana**
This is to follow the information rearchitecture work. This for the moment use the existing pages as is and we can in the future do more complete rewrites.

Doesn't really do much to Gateway because it's done in its own separate PR

<Explain your change!>

Did you sign your commit? [Instructions](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md#sign-your-commits)

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: app/_data/docs_nav_kuma_dev.yml [hunk 1]
```diff
@@ -126,21 +137,65 @@ items:
...
+      - text: Control Plane Configuration
+        url: /documentation/configuration/
+        items:
+          - text: Modifying the configuration
+            url: /documentation/configuration/#modifying-the-configuration
+          - text: Inspecting the configuration
+            url: /documentation/configuration/#inspecting-the-configuration
+          - text: Store
+            url: /documentation/configuration/#store
+  - title: Using Kuma
```
Comment (lukidzi) [inline app/_data/docs_nav_kuma_dev.yml:149]: Should we use?
```suggestion
  - title: Using {{site.mesh_product_name}}
```

File: app/_data/docs_nav_kuma_dev.yml [hunk 2]
```diff
@@ -117,6 +119,15 @@ items:
             url: /production/secure-deployment/certificates/
       - text: Kuma user interface
```
Comment (lukidzi) [inline app/_data/docs_nav_kuma_dev.yml:120]:
```suggestion
      - text: {{site.mesh_product_name}} user interface
```

Comment (bartsmykla) [reply app/_data/docs_nav_kuma_dev.yml:120]: I'm not sure we are able to use templates in this file

File: app/_data/docs_nav_kuma_dev.yml (hunk 1 above)
Comment (bartsmykla) [reply app/_data/docs_nav_kuma_dev.yml:149]: I'm not sure we are able to use templates in this file

Comment (lahabana) [reply app/_data/docs_nav_kuma_dev.yml:149]: Yes it doesn't we need to do a manual change in the sidebar overrides

**Review by lukidzi — APPROVED** [review]

**Pull request merged**